### PR TITLE
[Bounty] Stowaway RP Role

### PR DESCRIPTION
**Pull request by Neyusha**
# About The Pull Request

New RP role, adding the ability to play stowaways. All done as requested in the bounty #453 . With some of my additions.

Stowaways spawn in maintenance tunnels with an unknown ID card, a toolbelt, a flashlight and a radio. They are not visible on the crew list, no announcement of their arrival (when they enter a round), and they have no records.

They can be a traitor's target, they can be absolutely any in-game antagonist. 

Stowaways can be given a jobban, just like any other role.

By default, you can spawn in the game as a stowaway, but you cannot enter a round already in progress. To enter an already ongoing round, the admin should add a position in the admin panel.

The default maximum number of stowaways is 3, but as I wrote above - the admin can open new positions for that role.

According to the request in the bounty, when a stowaway enters the game, the stowaway gets a red sign informing them of the rules for playing for stowaways.

I have checked the code several times and it works fine. And if any thing seemed strange to you, I can change it.

# SCREENSHOTS
image
Manage Job Slots Admin Panel
image
image
Equipment
image
Crew Manifest
image
Jobban Panel
image

## Why It's Good For The Game

More nice roleplay situations, more noncon roleplay, more work for Security Officers.

## A Port?

No.

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog

:cl:Neyusha
add: Added stowaway role
/:cl:

### CODE REVIEW

File: code/__DEFINES/jobs.dm [hunk 1]
```diff
@@ -101,3 +102,4 @@
 #define JOB_DISPLAY_ORDER_BLUESHIELD 37
 #define JOB_DISPLAY_ORDER_BO 38
 #define JOB_DISPLAY_ORDER_PRISONER 39
+#define JOB_DISPLAY_ORDER_STOWAWAY 40
```
Comment (Neyusha) [inline code/__DEFINES/jobs.dm:105]: This allows to display stowaways at the bottom of the occupation list. Right after the prisoners.

**Comment by KathrinBailey** [comment]
Seems like it'd work better as a ghost role, but since you've already put the effort in, have you ever seen the miner conscription kit?
It gives you a card that, when used on your ID, adds an access to the ID.
Using that code you can make an item that, once certain conditions are met, will give access once applied to an ID.
You'd be able to make the stowaways get maints access for example, in exchange for putting so and so in something.
Should be super easy to make, you could even put the access cards on the black market uplink.

**Comment by Neyusha** [comment]
> Seems like it'd work better as a ghost role, but since you've already put the effort in, have you ever seen the miner conscription kit? It gives you a card that, when used on your ID, adds an access to the ID. Using that code you can make an item that, once certain conditions are met, will give access once applied to an ID. You'd be able to make the stowaways get maints access for example, in exchange for putting so and so in something. Should be super easy to make, you could even put the access cards on the black market uplink.

That's a great idea, but stowaways need access to the maints, otherwise they would have to hack literally every airlock just to walk through the maint tunnels. I think that's too painful for them :D

Without maint access, they can't even run away from the officers in the maint tunnels. 💀

**Comment by Cleric2145** [comment]
On delta station, Stowaways can sometimes spawn inside the "Syndicate Casino" room with no access to the door, leaving them stuck hacking the door or chopping through the door/walls. 

In my instance, this led to the research director hearing it, seeing me break out. He then went full sperg and called sec as he tried chasing me. I get it, but...it was 60 seconds into round >.>

**Comment by Neyusha** [comment]
> On delta station, Stowaways can sometimes spawn inside the "Syndicate Casino" room with no access to the door, leaving them stuck hacking the door or chopping through the door/walls.
> 
> In my instance, this led to the research director hearing it, seeing me break out. He then went full sperg and called sec as he tried chasing me. I get it, but...it was 60 seconds into round >.>
Fixed.

**Comment by LunarFleet** [comment]
For some reason, people spawning in as a stowaway can sometimes have their character slot completely broken, forcing them to remake the character. I am not quite sure _why_ this is happening, however, I have seen it happen twice already, involving a different player each time.

**Comment by BongaTheProto** [comment]
plz fix the merge conflicts

**Comment by SpookyTheFox** [comment]
I'm trying to get all the ghost roles tracked on the "check antags" screen like ashwalkers and space hotel staff, which would make it a lot easier to keep tabs on them from an admin perspective. Could you add the datum to track these?

**Pull request merged**